src/query/src/part_sort.rs

```diff
@@ -687,6 +694,12 @@ impl PartSortStream {
         cx: &mut Context<'_>,
     ) -> Poll<Option<datafusion_common::Result<DfRecordBatch>>> {
         loop {
+            // Early termination: if we've already produced enough rows,
+            // don't poll more input - just return
+            if matches!(self.limit, Some(0)) {
```

Comment:
should this be like `self.limit.map(|limit|limit<=self.produced).unwrap_or(false)` ?